lib/redis_memo/memoize_query/cached_select/bind_params.rb

```diff
@@ -2,47 +2,217 @@
 
 class RedisMemo::MemoizeQuery::CachedSelect
   class BindParams
+    def initialize(left = nil, right = nil, operator = nil)
+      @left = left
+      @right = right
+      @operator = operator
+    end
+
+    def union(other)
+      return unless other
+
+      self.class.new(self, other, __method__)
+    end
+
+    def product(other)
+      return unless other
+
+      self.class.new(self, other, __method__)
+    end
+
+    def should_cache?
+      plan!
+
+      if plan.model_attrs.empty? || plan.dependency_size_estimation.to_i > RedisMemo::DefaultOptions.max_query_dependency_size
+        return false
+      end
+
+      plan.model_attrs.each do |model, attrs_set|
```

Comment:
Maybe a micro-optimization, but I wonder if we can stop traversing through the AST if we hit a column that isn't memoized and exit early, rather than first having to `plan!` (though I guess the costly logic is in the `extract!`)

Reply:
I thought about that -- does not seem practical though, because we don't know the exact set of attrs (e..g the id,name attr set) until the full AST has been passed.

E.g. A set of attrs might become cachable later, even if it is not currently at the current AST node